Library/Homebrew/utils/curl.rb

```diff
@@ -328,30 +338,30 @@ def curl_http_content_headers_and_checksum(
         user_agent:        user_agent
       )
 
-      status_code = :unknown
-      while status_code == :unknown || status_code.to_s.start_with?("3")
-        headers, _, output = output.partition("\r\n\r\n")
-        status_code = headers[%r{HTTP/.* (\d+)}, 1]
-        location = headers[/^Location:\s*(.*)$/i, 1]
-        final_url = location.chomp if location
-      end
-
       if status.success?
+        parsed_output = parse_curl_output(output)
+        heads = parsed_output[:heads]
+
+        status_code = heads.last[:status_code] if heads.last.is_a?(Hash)
+        final_url = curl_response_last_location(heads)
+
+        headers = heads.last[:headers]
+        etag = headers["etag"][%r{^([wW]/)?"(([^"]|\\")*)"}, 2] if headers["etag"]
```

Comment:
No idea what this regex is doing or why we only care about the second capture group. Can it get extracted into a constant or variable with comment?

Reply:
This regex wasn't added in this PR (the pre-existing code can be found [here](https://github.com/Homebrew/brew/blob/baceee9d24f0a954bb2092272b9c3c834964b724/Library/Homebrew/utils/curl.rb#L350)) but it seems that the intention is to only capture the value within quotation marks. ETags have a format like `W/"<etag_value>"` or `"<etag_value>"` (e.g., `etag: "3147526947+gzip"` from https://www.example.com/), where a preceding `W/` indicates that a weak validator is used.

The only code in `brew` that uses the `:etag` field is `#curl_check_http_content`, so only using the value within quotation marks (i.e., ignoring the weak validator indicator) is probably intended to simplify the related comparison code:

```ruby
etag_match = details[:etag] &&
             details[:etag] == secure_details[:etag]
```

I'll move the regex into a constant with an explanatory comment before it.